app/presenters/results/pagination.rb

```diff
@@ -0,0 +1,41 @@
+module Results
+  module Pagination
+    PAGE_SIZE = 10
+
+    def total_records
+      raise NotImplementedError
+    end
+
+    def current_page
+      raise NotImplementedError
+    end
+
+    def total_pages
+      if total_records < page_size
+        1
+      elsif (total_records % page_size).zero?
+        total_records / page_size
+      else
+        (total_records / page_size) + 1
+      end
+    end
```

Comment:
Could this method be refactored to be a bit more readable, for example each condition defined as a method and use the method to switch on?

Reply:
I understand where are you coming from but the condition `total_records < page_size` is clear and I think there is no need to move to a method. Although the `(total_records % page_size).zero?` could be moved to a method as it is more trickier than the first one.

Does this looks better?

```ruby
  def total_pages
   if total_records < page_size
        1
      elsif multiple_of_page_size?
        total_records / page_size
      else
        (total_records / page_size) + 1
      end
   end

    def multiple_of_page_size?
      (total_records % page_size).zero?
    end
```

Btw this method was already there and we just moved from [here](https://github.com/moneyadviceservice/rad_consumer/pull/352/files#diff-2ec274c8a1d4f9569c910ba246df8ef5)

Reply:
Could also condense this down with some guard clauses:

```
def total_pages
  return 1 if total_records < page_size
  return (total_records / page_size) if multiple_of_page_size?
  (total_records / page_size) + 1
end
```

Reply:
I think Malcolm's suggestion is better. Also I would argue that just because a method has previously existed, does not mean that it is not a candidate for refactoring/improving.

Reply:
> I think Malcolm's suggestion is better

I also like it quite a log.

> Also I would argue that just because a method has previously existed, does not mean that it is not a candidate for refactoring/improving

Just to clarify, we didn't mean that, we never said that indeed. It's simply helpful to highlight, especially in a big PR like this, what has been actually introduced and what has simply been moved in order to better focus on code more relevant to the ticket. No doubt that any candidate for refactoring is a good candidate indeed.

Reply:
I apologise if I misunderstood that line; read within the context of the rest of the comment it had that implication.